common/remote_util.cc

```diff
@@ -47,10 +49,29 @@ void RemoteUtil::SetScpCommand(std::string scp_command) {
   scp_command_ = std::move(scp_command);
 }
 
+void RemoteUtil::SetSftpCommand(std::string sftp_command) {
+  sftp_command_ = std::move(sftp_command);
+}
+
 void RemoteUtil::SetSshCommand(std::string ssh_command) {
   ssh_command_ = std::move(ssh_command);
 }
 
+// static
+std::string RemoteUtil::ScpToSftpCommand(std::string scp_command) {
+  // "scp", "winscp.exe", "C:\path\to\scp", "/scppath/scp --foo" etc.
+  size_t pos = 0;
+  while ((pos = scp_command.find("scp", pos)) != std::string::npos) {
+    const char next_ch = scp_command[pos + 3];
```

Comment:
This goes out of bounds for `C:\path\to\scp`.

Reply:
It may access the string at scp_command.size(), which is well defined in C++11: "If pos is equal to the string length, the function returns a reference to the null character that follows the last character in the string (which should not be modified)." - https://cplusplus.com/reference/string/string/operator[]/. I'll add a comment.